### PR TITLE
Use irep id defs for all #comments

### PR DESCRIPTION
**Pull request by peterschrammel**

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 12e970bc5d479d57b3f0a9c6649bf87aa34ff98f).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/79643317

**Review by kroening — APPROVED** [review]

**Review by chrisr-diffblue — APPROVED** [review]

File: src/cpp/cpp_declarator_converter.cpp [hunk 1]
```diff
@@ -258,7 +258,7 @@ void cpp_declarator_convertert::combine_types(
         if(decl_parameter.type()!=symbol_parameter.type())
         {
           // The 'this' parameter of virtual functions mismatches
-          if(i!=0 || !symbol_code_type.get_bool("#is_virtual"))
+          if(i != 0 || !symbol_code_type.get_bool(ID_C_is_virtual))
```
Comment (martin-cs) [inline src/cpp/cpp_declarator_converter.cpp:261]: Should the C++ front-end be generating ID_C_* things for C++ features?

File: src/cpp/cpp_typecheck_compound_type.cpp [hunk 2]
```diff
@@ -888,9 +888,9 @@ void cpp_typecheckt::typecheck_friend_declaration(
     // typecheck ftype
 
     // TODO
-//    typecheck_type(ftype);
-//    assert(ftype.id()==ID_symbol);
-//    symbol.type.add("#friends").move_to_sub(ftype);
+    //    typecheck_type(ftype);
+    //    assert(ftype.id()==ID_symbol);
+    //    symbol.type.add("ID_C_friends").move_to_sub(ftype);
```
Comment (martin-cs) [inline src/cpp/cpp_typecheck_compound_type.cpp:893]: I'm not sure if updating commented out TODO code is better or worse than not updating it.  Hmmm... I think updating is probably the right thing to do.

**Review by martin-cs — APPROVED** [review]
It feels like some kind of documentation for these, note that they are comments, etc. might be a good thing.

File: src/cpp/cpp_declarator_converter.cpp (hunk 1 above)
Comment (smowton) [reply src/cpp/cpp_declarator_converter.cpp:261]: That `_C_` isn't C the language, it's "Comment". C the language seems to usually use a lowercase 'c' in irep identifiers.

**Comment by peterschrammel** [comment]
> It feels like some kind of documentation for these, note that they are comments, etc. might be a good thing.

Correct, looking at this table http://cprover.diffblue.com/irep__ids_8cpp.html#ac0dc1891d23310a88b023bf30cce5287a9c62234e0e0df1288e8a323e2f9e5cca there is a yawning void in the documentation that would be worth filling.

**Pull request merged**